### PR TITLE
Update code for update movie by calling API 

### PR DESCRIPTION
**Pull request by jinety**

### CODE REVIEW

File: big-practice/src/js/pages/dashboard.page.js [hunk 1]
```diff
@@ -42,23 +43,34 @@ class Dashboard {
...
-  //  *
-  //  * @param {element} item - Table update button
-  //  */
-  // tableUpdateBtn(item) {
+  /**
+   * Modal form update will appear when clicking the update button in the table
+   *
+   * @param {element} item - Table update button
+   */
+  // async tableUpdateBtn(item) {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: Duplicated code

File: big-practice/src/js/pages/dashboard.page.js [hunk 2]
```diff
@@ -42,23 +43,34 @@ class Dashboard {
...
+  //   ModalHelper.showModal(this.modalForm);
+  //   const movieData = await apiService.get(`${MOVIES_API}/${movieId}`);
+
+  //   this.nameMovieInput.value = movieData.name;
+  //   this.directorInput.value = movieData.director;
+  //   this.nationInput.value = movieData.nation;
+  //   this.form.setAttribute('data-id', movieId);
   // }
 
+  async tableUpdateBtn(item) {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: What does tableUpdateBtn mean? Function should be started with a verb and describe what it really do

File: big-practice/src/js/pages/dashboard.page.js [hunk 3]
```diff
@@ -131,41 +161,48 @@ class Dashboard {
...
+        return;
+      }
+
+      const movieList = await apiService.get(`${MOVIES_API}?name=${data.name}`);
+      const moviesDoNotExist = movieList.length === 0
+        || movieList[0].id === parseInt(formMovieId, 10);
+
+      // Check if the movie exists or not
+      if (moviesDoNotExist) {
+        await apiService.put(`${MOVIES_API}/${formMovieId}`, data);
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: As create movie feature, update movie should not be render all table.

File: big-practice/src/js/pages/dashboard.page.js [hunk 4]
```diff
@@ -222,11 +259,13 @@ class Dashboard {
   //   hideModal(modalWarning);
   // });
 
-  // // Movie will be updated when the update button is clicked
-  // modalFormUpdateBtn.addEventListener('click', () => {
-  //   handleUpdateForm();
-  //   hideModal(modalWarning);
-  // });
+  handleModalFormUpdateBtn() {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: handleModalFormUpdateBtn is a complicated name, could you use another simple name?

File: big-practice/src/js/pages/dashboard.page.js [hunk 5]
```diff
@@ -131,41 +161,48 @@ class Dashboard {
...
+        DocumentHelper.showErrorMessage(this.nameMovieInput, validate.errors.name);
+        DocumentHelper.showErrorMessage(this.directorInput, validate.errors.director);
+        DocumentHelper.showErrorMessage(this.nationInput, validate.errors.nation);
+
+        return;
+      }
+
+      const movieList = await apiService.get(`${MOVIES_API}?name=${data.name}`);
+      const moviesDoNotExist = movieList.length === 0
+        || movieList[0].id === parseInt(formMovieId, 10);
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: Should parseInt in helper

File: big-practice/src/js/helpers/parse.helper.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+export class Parse {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/helpers/parse.helper.js]: What is type of Parse?

File: big-practice/src/js/pages/dashboard.page.js [hunk 7]
```diff
@@ -84,6 +85,15 @@ class Dashboard {
       });
 
       this.tableBody.innerHTML = tableTemplate;
+      const updateButtons = document.querySelectorAll('.table .table-update-btn');
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: From line 88 ~ 96 -> Duplicated code

File: big-practice/src/js/pages/dashboard.page.js [hunk 8]
```diff
@@ -181,7 +211,7 @@ class Dashboard {
   /**
    * Handle modal appearance when user clicks add button
    */
-  handleLayoutMainAddBtn() {
+  handleLayoutMainAdd() {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: Func name does not make sense

File: big-practice/src/js/pages/dashboard.page.js [hunk 9]
```diff
@@ -193,7 +223,7 @@ class Dashboard {
   /**
    * Handle creating new movie when user clicks create movie button
    */
-  handleModalFormCreateBtn() {
+  handleModalFormCreate() {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: Func name does not make sense

File: big-practice/src/js/pages/dashboard.page.js [hunk 10]
```diff
@@ -203,7 +233,7 @@ class Dashboard {
   /**
    * Handle modal exit when user clicks cancel button
    */
-  handleModalFormCancelBtn() {
+  handleModalFormCancel() {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: Func name does not make sense

File: big-practice/src/js/pages/dashboard.page.js [hunk 11]
```diff
@@ -222,11 +252,13 @@ class Dashboard {
   //   hideModal(modalWarning);
   // });
 
-  // // Movie will be updated when the update button is clicked
-  // modalFormUpdateBtn.addEventListener('click', () => {
-  //   handleUpdateForm();
-  //   hideModal(modalWarning);
-  // });
+  handleModalFormUpdate() {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: Func name does not make sense

File: big-practice/src/js/pages/dashboard.page.js [hunk 12]
```diff
@@ -131,41 +141,61 @@ class Dashboard {
...
+            await this.handleTableUpdate(item);
+            DocumentHelper.hideElement(this.modalFormCreateBtn);
+            DocumentHelper.showElement(this.modalFormUpdateBtn);
+          });
+        });
+
+        // await this.handleRenderTable();
+        ModalHelper.hideModal(this.modalForm);
+      } else {
+        DocumentHelper.showErrorMessage(this.nameMovieInput, MESSAGES.exist);
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: MESSAGES.exist => Which thing exist?

File: big-practice/src/js/pages/dashboard.page.js [hunk 13]
```diff
@@ -131,41 +141,61 @@ class Dashboard {
...
+
+        updateButtons.forEach((item) => {
+          item.addEventListener('click', async () => {
+            await this.handleTableUpdate(item);
+            DocumentHelper.hideElement(this.modalFormCreateBtn);
+            DocumentHelper.showElement(this.modalFormUpdateBtn);
+          });
+        });
+
+        // await this.handleRenderTable();
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: Remove unused code

File: big-practice/src/js/pages/dashboard.page.js [hunk 14]
```diff
@@ -131,41 +141,61 @@ class Dashboard {
...
+      const movieList = await apiService.get(`${MOVIES_API}?name=${data.name}`);
+      const moviesDoNotExist = movieList.length === 0
+        || movieList[0].id === Parse.parseInt(formMovieId);
+
+      // Check if the movie exists or not
+      if (moviesDoNotExist) {
+        const rowUpdate = document.querySelector(`[data-id="${formMovieId}"]`);
+        const movieUpdate = await apiService.put(`${MOVIES_API}/${formMovieId}`, data);
+
+        rowUpdate.innerHTML = MovieTemplate.renderTableRow(movieUpdate);
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: Noun should be the last word. Ex: Black hat, NOT Hat Black.

This happens frequently on your code, ex: Line 176, 177, etc. Pls check around

File: big-practice/src/js/pages/dashboard.page.js [hunk 15]
```diff
@@ -131,41 +141,61 @@ class Dashboard {
...
+      if (!validate.isValid) {
+        DocumentHelper.showErrorMessage(this.nameMovieInput, validate.errors.name);
+        DocumentHelper.showErrorMessage(this.directorInput, validate.errors.director);
+        DocumentHelper.showErrorMessage(this.nationInput, validate.errors.nation);
+
+        return;
+      }
+
+      const movieList = await apiService.get(`${MOVIES_API}?name=${data.name}`);
+      const moviesDoNotExist = movieList.length === 0
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: moviesDoNotExist does not make sense. Boolean variable should start with is, has or enable/disable. Ex: isValidUser

File: big-practice/src/js/pages/dashboard.page.js (hunk 1 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

File: big-practice/src/js/pages/dashboard.page.js (hunk 2 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

File: big-practice/src/js/pages/dashboard.page.js (hunk 3 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

File: big-practice/src/js/pages/dashboard.page.js (hunk 4 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

File: big-practice/src/js/pages/dashboard.page.js (hunk 5 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

File: big-practice/src/js/helpers/parse.helper.js (hunk 6 above)
Comment (jinety) [reply big-practice/src/js/helpers/parse.helper.js]: Fixed

File: big-practice/src/js/pages/dashboard.page.js (hunk 7 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

File: big-practice/src/js/pages/dashboard.page.js (hunk 8 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

File: big-practice/src/js/pages/dashboard.page.js (hunk 9 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

File: big-practice/src/js/pages/dashboard.page.js (hunk 10 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

File: big-practice/src/js/pages/dashboard.page.js (hunk 11 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

File: big-practice/src/js/pages/dashboard.page.js (hunk 12 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

File: big-practice/src/js/pages/dashboard.page.js (hunk 13 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

File: big-practice/src/js/pages/dashboard.page.js (hunk 14 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

File: big-practice/src/js/pages/dashboard.page.js (hunk 15 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

File: big-practice/src/js/pages/dashboard.page.js [hunk 16]
```diff
@@ -42,22 +44,35 @@ class Dashboard {
...
-  //     nameMovieInput.value = movieData.name;
-  //     directorInput.value = movieData.director;
-  //     nationInput.value = movieData.nation;
-  //     form.setAttribute('data-id', movieId);
-  //   });
-  // }
+  /**
+   * Query to all update buttons in the table
+   */
+  queryUpdateButtons() {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: Wrong purpose, Query means you only query an element, but you also add event and handle

File: big-practice/src/js/pages/dashboard.page.js [hunk 17]
```diff
@@ -181,9 +207,10 @@ class Dashboard {
   /**
    * Handle modal appearance when user clicks add button
    */
-  handleLayoutMainAddBtn() {
+  showModalFormCreateMovie() {
     // Popup to add user when clicking on Add button.
     this.layoutMainAddBtn.addEventListener('click', () => {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: layoutMain??? Why we need this postfix?

File: big-practice/src/js/pages/dashboard.page.js [hunk 18]
```diff
@@ -193,7 +220,7 @@ class Dashboard {
   /**
    * Handle creating new movie when user clicks create movie button
    */
-  handleModalFormCreateBtn() {
+  createNewMovie() {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: Main target of this function should be add an event for create button, not create new movie. The same on updateMovie

File: big-practice/src/js/pages/dashboard.page.js (hunk 16 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

File: big-practice/src/js/pages/dashboard.page.js (hunk 17 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

File: big-practice/src/js/pages/dashboard.page.js (hunk 18 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Fixed

**Pull request merged**